### PR TITLE
PackageReference Migration

### PR DESCRIPTION
**Pull request by brthor**
WIP PackageReference, won't merge until we figure out what the plan is for imports.

We can use this branches bits to test PackageReference in the meantime though.

@livarcocc @Sridhar-MS @eerhardt @srivatsn

### CODE REVIEW

File: src/Microsoft.DotNet.ProjectJsonMigration/MigrationNuGetFrameworkExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.ProjectJsonMigration
+{
+    public static class MigrationNuGetFrameworkExtensions
+    {
+        public static string GetMSBuildCondition(this NuGetFramework framework)
+        {
+            return $" '$(TargetFrameworkIdentifier),Version=$(TargetFrameworkVersion)' == '{framework.DotNetFrameworkName}' ";
```
Comment (eerhardt) [inline src/Microsoft.DotNet.ProjectJsonMigration/MigrationNuGetFrameworkExtensions.cs]: I think you can just say `'$(TargetFramework)' == '{framework.ShortName}'`

/cc @nguerrera - to confirm if this is true.

File: src/Microsoft.DotNet.ProjectJsonMigration/Rules/TemporaryMutateProjectJsonRule.cs [hunk 2]
```diff
@@ -39,8 +40,8 @@ public void Apply(MigrationSettings migrationSettings, MigrationRuleInputs migra
             }
 
             var json = CreateDestinationProjectFile(sourceProjectFile, destinationProjectFile);
-            InjectSdkReference(json, ConstantPackageNames.CSdkPackageName, migrationSettings.SdkPackageVersion);
             RemoveRuntimesNode(json);
+            RemoveAllDependencies(json);
```
Comment (eerhardt) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/TemporaryMutateProjectJsonRule.cs]: Why?  I would think now we wouldn't need to mutate the project.json at all anymore.

Comment (brthor) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/TemporaryMutateProjectJsonRule.cs]: That's a good point, I was thinking that we still need it for tools, but that is not going to work

File: src/Microsoft.DotNet.ProjectJsonMigration/MigrationNuGetFrameworkExtensions.cs (hunk 1 above)
Comment (brthor) [inline src/Microsoft.DotNet.ProjectJsonMigration/MigrationNuGetFrameworkExtensions.cs]: Is this true @nguerrera? Will make the change

Comment (nguerrera) [inline src/Microsoft.DotNet.ProjectJsonMigration/MigrationNuGetFrameworkExtensions.cs]: As long as you put the same spelling in TargetFrameworks, then yes, and that is the idiomatic way within a project.

Comment (brthor) [inline src/Microsoft.DotNet.ProjectJsonMigration/MigrationNuGetFrameworkExtensions.cs]: 👍

Comment (brthor) [inline src/Microsoft.DotNet.ProjectJsonMigration/MigrationNuGetFrameworkExtensions.cs]: changed.

**Comment by brthor** [comment]
test Windows_NT x64 Release Build please

**Pull request merged**